onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -266,12 +270,22 @@ void TreeEnsembleCommon<ITYPE, OTYPE>::compute_agg(const Tensor* X, Tensor* Z, T
           agg.ProcessTreeNodePrediction1(score, *ProcessTreeNodeLeave(roots_[j], x_data));
       } else {
         std::vector<ScoreValue<OTYPE>> scores_t(n_trees_, {0, 0});
+#if defined(USE_THREADPOOL)
```

Comment:
Can you make a runtime check for n_trees > some threshold instead to decide if calling TryBatchParallelFor is needed?

Reply:
This test is done just above. It kept this switch to be able to compare with the old implementation. I removed it.